cloud_governance/policy/ec2_idle.py

```diff
@@ -0,0 +1,132 @@
+from datetime import datetime, timedelta
+
+from cloud_governance.common.aws.cloudwatch.cloudwatch_operations import CloudWatchOperations
+from cloud_governance.common.logger.init_logger import logger
+from cloud_governance.zombie_non_cluster.run_zombie_non_cluster_policies import NonClusterZombiePolicy
+
+
+class EC2Idle(NonClusterZombiePolicy):
+
+    TRIGGER_STOP_INSTANCE_PERIOD = 2
+    STOP_INSTANCE_PERIOD = 4
+    CPU_UTILIZATION_PERCENTAGE = 5
+    NETWORK_IN_BYTES = 5000
+    NETWORK_OUT_BYTES = 5000
+
+    def __init__(self):
+        super().__init__()
+        self._cloudwatch = CloudWatchOperations(region=self._region)
+
+    def run(self):
+        """
+        This method list all stopped instances for more than 30 days and terminate if dry_run no
+        @return:
+        """
+        return self.__fetch_stop_instance()
+
+    def __get_metrics_average(self, metric_list: list, metric_period: int):
+        """
+        This method calculate the average of the metrics
+        @param metric_list:
+        @param metric_period:
+        @return:
+        """
+        metrics = []
+        for metric in metric_list:
+            metrics.append(sum(metric['Values'])/metric_period)
+        return metrics
+
+    def __get_metrics_from_cloud_watch(self, instance_id: str, instance_period: int):
+        """
+        This method extracts the logs from the cloudwatch
+        @param instance_id:
+        @param instance_period:
+        @return:
+        """
+        start_time, end_time = datetime.now() - timedelta(days=instance_period), datetime.now()
+        metrics = self._cloudwatch.get_metric_data(start_time=start_time, end_time=end_time, resource_id=instance_id,
+                                                   resource_type='InstanceId', namespace='AWS/EC2',
+                                                   metric_names={'CPUUtilization': 'Percent', 'NetworkIn': 'Bytes', 'NetworkOut': 'Bytes'},
+                                                   statistic='Average')
+        return metrics['MetricDataResults']
+
+    def __get_proposed_metrics(self, metrics: list, metric_period: int):
+        """
+        This method return the metrics
+        @param metrics:
+        @param metric_period:
+        @return:
+        """
+        return self.__get_metrics_average(metric_list=metrics, metric_period=metric_period)
+
+    def __get_time_difference(self, launch_time: datetime):
+        """
+        This method return the difference of datetime
+        @param launch_time:
+        @return:
+        """
+        end_time = datetime.now()
+        return (end_time - launch_time.replace(tzinfo=None)).days
+
+    def __fetch_stop_instance(self):
```

Comment:
Why we need to fetch stop instance for idle

Reply:
No need, I just use the same code style of ec2_stop.

Reply:
Please fix it and leave here only the methods that related to ec2 idle